codegen/src/main/java/software/amazon/awssdk/codegen/poet/rules/EndpointResolverInterceptorSpec.java

```diff
@@ -125,6 +128,8 @@ public TypeSpec poetSpec() {
                                       .addAnnotation(SdkInternalApi.class)
                                       .addSuperinterface(ExecutionInterceptor.class);
 
+        b.addField(logger());
```

Comment:
In general, wondering if we should add logic to only generate operation context params if they exist, potentially including the logger dependency. Do not have time to check exactly what is generated under which circumstances for this PR.

Reply:
Added removing empty methods to follow up task